tsl/src/continuous_aggs/drop.c

```diff
@@ -15,9 +15,38 @@
 
 #include "create.h"
 
+/* drop chunks from the materialization hypertable that fall within the time
+ * range.
+ */
+static void
+cagg_drop_mat_chunks(Oid mattable_relid, Datum older_than_datum, Datum newer_than_datum,
+					 Oid older_than_type, Oid newer_than_type, int32 log_level)
+{
+	uint64 i = 0, num_chunks = 0;
+	Chunk *matchunks;
+
+	matchunks = ts_chunk_get_chunks_in_time_range(mattable_relid,
+												  older_than_datum,
+												  newer_than_datum,
+												  older_than_type,
+												  newer_than_type,
+												  "cagg_drop_mat_chunks",
+												  CurrentMemoryContext,
+												  &num_chunks,
+												  false);
+	// TODO any reason why this should be anything other than DROP_CASCADE
+	for (; i < num_chunks; i++)
+	{
+		ts_chunk_drop(matchunks + i, DROP_CASCADE, log_level);
```

Comment:
DROP CASCADE is the more dangerous option so my question is can this be DROP_RESTRICT? but really can't we pass the same option that was part of the statement that caused this in the first place?

Reply:
I would actually go further and suggest we avoid `CASCADE` even when there are dependent objects that should also be removed. I think it is better to be explicit and first remove the dependent objects in that case---then one knows exactly what is going on and what the result is. It also protects against changes that might build new dependencies where `CASCADE` is not suitable anymore.

Minor nit:

I think  `&matchunks[i]` is more readable.

Reply:
Well there may be user-defined dependencies here so I think if cascade was used in the original statement it's wise to pass it down and use it here. Otherwise, all internal dependencies (I don't think there are any) should be processed explicitly and restrict used here.

Reply:
modified to pass on the cascade behavior.